src/components/TextArea/TextArea.st.css

```diff
@@ -0,0 +1,94 @@
+:import {
+    -st-from: "../../common/formatters.st";
+    -st-named: applyUnit, applyOpacity, font, fallback, color, zeroAsTrue;
+}
+
+:import {
+    -st-from: "../../common/palette.st.css";
+    -st-named: errorColor, successColor;
+}
+
+:vars {
+    /*Defaults*/
+    DefaultBorderColor: color-5;
+    DefaultBackgroundColor: color-1;
+    DefaultTextColor: color-5;
+    DefaultPlaceholderColor: color-3;
+    DefaultFocusFontColor: color-1;
+    DefaultDisabledBorderColor: color-3;
+    DefaultFocusFontColor: color-1;
+    DefaultFont: "{theme: 'Body-M', size: '16px', lineHeight: '1.5em'}";
+
+    /*Defaults params goes here*/
+}
+
+:vars {
+    /*Overrides*/
+    BorderColor: --overridable;
+    BackgroundColor: --overridable;
+    TextColor: --overidable;
+    PlaceholderColor: --overidable;
+    TextFont: --overidable;
+    ErrorBorderColor: --overidable;
+    SuccessBorderColor: --overidable;
+    DisabledBorderColor: --overidable;
+    /*Deceleration of overridable placeholders*/
+}
+
+:vars {
+    borderColor: color(fallback(value(BorderColor), value(DefaultBorderColor)));
+    backgroundColor: color(fallback(value(BackgroundColor), value(BackgroundColor)));
+    textColor: color(fallback(value(TextColor), value(DefaultTextColor)));
+    placeholderColor: color(fallback(value(PlaceholderColor), value(DefaultPlaceholderColor)));
+}
+
+.root {
+    -st-states: theme(enum(line, box)), disabled, error, success;
+    position: relative;
+    width: 280px;
+    height: 92px;
+}
+
+.root:theme(line) .textArea {
+    border-top: none;
+    border-left: none;
+    border-right: none;
+}
+
+.textArea {
+    width: 100%;
+    height: 100%;
+    resize: none;
+    padding: 8px 12px;
+    font: fallback(font(value(TextFont)), font(value(DefaultFont)));
+    box-sizing: border-box;
+    color: value(textColor);
+    border-color: value(borderColor);
+    background-color: value(backgroundColor);
+}
+
+.textArea::placeholder,
+.textArea::-webkit-input-placeholder
+{
+    color: value(placeholderColor);
+}
+
+.errorIcon {
```

Comment:
Note that there should be an extra padding when the icon appears (see the zeroheight spec, "Filled w. Error" example).
So I guess this should probably be added as a stylable state to the root, and added to the textarea when the icon appears

Reply:
In addition, we need to add rtl direction for this icon.
What I suggest you can do, is to have something like this:
```jsx
<div {...styles('root', {...}, this.props} >
  ...
  <div className={styles.errorIconWrapper}>
    <Tooltip><ErrorIcon /></Tooltip>
  </div>
  ...
</div>
```
```css
.errorIconWrapper {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  display: flex;
  justify-content: flex-end;
}
```